Blog on Rust (#68)
data-umbrella/data-umbrella.github.io: _posts/2022-11-15-rust-blog-tutorial.md

---
layout: post
title:  Tutorial on Intro to Rust Programming
description: An introduction to Rust Programming for complete beginners
date:   2022-11-15 15:01:35 +0300
image:  '/images/blogs/bruno-rust-blog.png'
tags:   [rust]
---

Summary posted by: [Sangam SwadiK](https://www.linkedin.com/in/sangam-swadi-k/)

> ## Event
> An introduction to Rust Programming for complete beginners, we will go over what is Rust, how to install it, how to use it online, basic structure of a program, core concepts of Rust and a little CLI program.
> 

Reviewer: ```suggestion
## Intro
An introduction to Rust Programming for complete beginners. This 2-hour video tutorial covers:  
- what is Rust
- how to install Rust
- how to use Rust online
- basic structure of a Rust program
- core concepts of Rust
- a little CLI (command line interface) program
- Q&A 
```

## Video
<p>
<iframe src="https://www.youtube.com/embed/7E8nLExn3WI" loading="lazy" frameborder="0" allowfullscreen></iframe>
</p>

## Resources
- [Event transcripts and notes](https://github.com/rochacbruno/rust-intro/blob/wip/script.md)

> - Slides: [slides](https://github.com/rochacbruno/rust-intro)
> - Data science libraries in Rust: [https://www.arewelearningyet.com/](https://www.arewelearningyet.com/)

Reviewer: ```suggestion
- GitHub repo: [rust-intro](https://github.com/rochacbruno/rust-intro)
- Data science libraries in Rust: [https://www.arewelearningyet.com/](https://www.arewelearningyet.com/)
```

- [01:34:36](https://www.youtube.com/watch?v=7E8nLExn3WI&t=5676s) Borrowing
- [01:37:37](https://www.youtube.com/watch?v=7E8nLExn3WI&t=5857s) Console Input
- [01:40:18](https://www.youtube.com/watch?v=7E8nLExn3WI&t=6018s) DEMO ERROR ignore this part :)
- [01:43:02](https://www.youtube.com/watch?v=7E8nLExn3WI&t=6182s) Read user input
- [01:45:22](https://www.youtube.com/watch?v=7E8nLExn3WI&t=6322s) Combinators
- [01:46:01](https://www.youtube.com/watch?v=7E8nLExn3WI&t=6361s) Result type
- [01:50:00](https://www.youtube.com/watch?v=7E8nLExn3WI&t=6600s) External Dependency
- [01:52:07](https://www.youtube.com/watch?v=7E8nLExn3WI&t=6727s) Random numbers
- [01:54:38](https://www.youtube.com/watch?v=7E8nLExn3WI&t=6878s) Type coercion
- [01:59:09](https://www.youtube.com/watch?v=7E8nLExn3WI&t=7149s) Complete program
- [02:01:47](https://www.youtube.com/watch?v=7E8nLExn3WI&t=7307s) Q & A

## About the Speaker
### Bio
Bruno is a Software Engineer who works developing the Red Hat Ansible Platform, he is member of the Python Software Foundation, creator of Dynaconf settings manager and a Rust enthisiast for 5+ years.

### Connect with the Speaker, Bruno Rocha

> - LinkedIn: [Bruno Rocha](https://www.linkedin.com/in/rochacbruno)

Reviewer: ```suggestion
- YouTube: [@rochacbruno](https://www.youtube.com/@rochacbruno)
- LinkedIn: [Bruno Rocha](https://www.linkedin.com/in/rochacbruno)
```

- Twitter: [@rochacbruno](https://twitter.com/rochacbruno)
- GitHub: [@rochacbruno](https://github.com/rochacbruno)

> ## Key Links
> - [Meetup Event](https://www.meetup.com/data-umbrella/events/289308172/)
> - [Video](https://www.youtube.com/watch?v=7E8nLExn3WI)
> - [GitHub repo](https://github.com/rochacbruno/rust-intro)

Reviewer: ```suggestion
```

Reviewer: delete these links, they are a duplicate of above.

---
layout: post
title:  Tutorial on Intro to Rust Programming
description: An introduction to Rust Programming for complete beginners
date:   2022-11-15 15:01:35 +0300
image:  '/images/blogs/bruno-rust-blog.png'
tags:   [rust]
---

Summary posted by: [Sangam SwadiK](https://www.linkedin.com/in/sangam-swadi-k/)

## Intro
An introduction to Rust Programming for complete beginners. This 2-hour video tutorial covers:  
- what is Rust
- how to install Rust
- how to use Rust online
- basic structure of a Rust program
- core concepts of Rust
- a little CLI (command line interface) program
- Q&A 

> ## Video

Reviewer: ```suggestion

## Video
```
